Detectors/TPC/qc/include/TPCQC/Clusters.h

```diff
@@ -0,0 +1,69 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// @file   Clusters.h
+/// @author
+///
+
+#ifndef AliceO2_TPC_CLUSTERS_H
+#define AliceO2_TPC_CLUSTERS_H
+
+//root includes
+#include "TCanvas.h"
+
+//o2 includes
+#include "DataFormatsTPC/Defs.h"
+#include "DataFormatsTPC/ClusterNative.h"
+#include "TPCBase/CalDet.h"
+#include "TPCBase/Sector.h"
+
+namespace o2
+{
+namespace tpc
+{
+namespace qc
+{
+
+/// Keep QC information for Clusters related observables
+///
+class Clusters
+{
+ public:
+  Clusters() = default;
+
+  bool processCluster(o2::tpc::ClusterNative const& cluster, o2::tpc::Sector const sector, const int row);
+
+  void analyse();
+
+  void dumpToFile(std::string filename);
```

Comment:
I would prefer `std::string_view` and `#include <string_view>`.

Reply:
What I wanted to achieve is that the user can give a name for the output to the runClusters.C so that they can either type `name.root` or simply `name`. Then they receive a `name.root` file containing the data members of the CalPad objects and another file called `name_canvas.root` containing the canvases that were drawn by the painter. 
As far as I see this was easy with std:: string objects but is not possible with std::string_view objects. Imho std::string_view objects can not be manipulated. Also functions like ends_with() that would simplify the code e.g. `if (filename.find(".root") != std::string::npos) {...` only work in C++20. 
I would suggest to stick with std::string objects or am I missing something?

Reply:
See below in Clusters.cxx